src/main/java/com/microsoft/jenkins/azuread/AzureAdUser.java

```diff
@@ -108,6 +126,7 @@ public boolean equals(Object o) {
         if (familyName != null ? !familyName.equals(that.familyName) : that.familyName != null) return false;
         if (uniqueName != null ? !uniqueName.equals(that.uniqueName) : that.uniqueName != null) return false;
         if (tenantID != null ? !tenantID.equals(that.tenantID) : that.tenantID != null) return false;
+        if (groupOIDs != null ? !groupOIDs.equals(that.groupOIDs) : that.groupOIDs != null) return false;
```

Comment:
Is it possible that the `groupOIDs` has same elements with different orders?

Reply:
You're right with the comment. I'm not sure the lists are always in the same sequence. Provided a fix for this.